frontend/docs/index.mdx

```diff
@@ -0,0 +1,268 @@
+
+We use jest and puppeteer for the end to end tests. There are some mocks for the requests in the puppeteer/mocks folder. Every slash (/) and params (? &) are replaced by a hyphen in the file name. i.e:
+api-v3-contexts-year-2015 will mock api/v3/contexts?year=2015
+
+The tests have to be run while the js server is running
+
+Run all tests with:
+```
+yarn test
+```
+
+Run only unit tests:
+```
+yarn unit
+```
+
+Run only end to end:
+```
+yarn e2e
+```
+
+If you need to change some snapshot due to changes in the page and the test is failing:
+
+```
+yarn e2e -- -u
+```
+
+### API
+
+```
+RAILS_ENV=test rake db:drop db:create db:structure:load
+bundle exec rspec spec
+```
+
+You may also want to know about API [gold master tests](docs-backend-gold-master-tests) and [load tests](docs-backend-load-tests).
+
+### Frontend
+
+Running `yarn test` will launch jest testing environment and run all tests that match `*.test.js` filename inside the `frontend/scripts` folder.
+
+## Deployment
+
+We use [Capistrano](http://capistranorb.com/) as a deployment tool, which deploys both API and frontend application simultaneously.
+
+To deploy, simply use:
+
+```
+cap <staging|production> deploy
+```
+
+## Git hooks
+
+This project includes a set of git hooks that you may find useful
+- Run `bundle install` when loading `Gemfile.lock` modifications from remotes
+- Run `yarn install` when loading `frontend/package.json` modifications from remotes
+- Receive a warning when loading `.env.sample` or `frontend/.env.sample` modifications from remotes
+- Run `Rubocop` and `JS Lint` before committing
+
+To enable then, simply execute once: `bin/git/init-hooks`
+
+## Documentation
+
+### API
+
+#### Database
+- [full schema](https://vizzuality.github.io/trase/all_tables/)
+- ["blue tables" schema](https://vizzuality.github.io/trase/blue_tables/index.html)
+- [how to regenerate docs](docs-backend-automated-documentation)
+
+#### More
+- [Cache management](docs-backend-cache-management)
+- [Background jobs](docs-backend-background-jobs)
+- [Data update](docs-backend-data-update)
+- [Copying databases between instances](docs-backend-data-export-import)
+
+### Frontend
+
+#### Lexicon
+
+```
++-------+             +-------+
+|       |             |       |
+|       |             |       |
++-------+ ---\        |       |
+| node  |     \-------+-------+
++-------+--\  link    | node  |
+|       |   \         |       |
+|       |    \--------+-------+
+|       |             |       |
++-------+             +-------+
+  column                column
+
+```
+
+### Components playground
+
+Online in: http://trase.surge.sh
+
+#### Run locally
+- Run `yarn docz:dev`
+It will be served on localhost:8081
+
+#### Build
+- Run `yarn docz:build`
+
+#### Deploy
+- Install surge `yarn add -g surge`
+- Run build
+- Go to docs/dist folder
+- Run surge `surge . trase.surge.sh`
+- If asked for login use trase.vizzuality@gmail.com
```

Comment:
Perhaps there's a way to automate this using travis: https://docs.travis-ci.com/user/deployment/surge/